subprojects/core/src/main/java/org/gradle/process/internal/DefaultJavaForkOptions.java

```diff
@@ -215,4 +251,13 @@ public boolean isCompatibleWith(JavaForkOptions options) {
                 && containsAll(getEnvironment(), options.getEnvironment())
                 && getBootstrapClasspath().getFiles().containsAll(options.getBootstrapClasspath().getFiles());
     }
+
+    private static boolean hasJvmArgumentProviders(JavaForkOptions forkOptions) {
+        if (!(forkOptions instanceof DefaultJavaForkOptions)) {
+            return false;
+        }
```

Comment:
Is this to support custom implementations of `JavaForkOptions`? Do we want to do that? `JavaForkOptions` states that it `@HasInternalProtocol`, so I'm not even sure what it would mean if we had custom implementations.

Reply:
We have `ImmutableJavaForkOptions`. `ImmutableJavaForkOptions` do not have any argument providers (`getJvmArgumentProviders` throws an `UnsupportedOperationException`).
This is why I do this check only for `DefaultJavaForkOptions`.